src/ol/disposable.js

```diff
@@ -0,0 +1,30 @@
+goog.provide('ol.Disposable');
+
+/**
+ * Objects that need to clean up after themselves.
+ * @constructor
+ */
+ol.Disposable = function() {};
+
+/**
+ * The object has already been disposed.
+ * @type {boolean}
+ * @private
+ */
+ol.Disposable.prototype.disposed_ = false;
+
+/**
+ * Clean up.
+ */
+ol.Disposable.prototype.dispose = function() {
```

Comment:
@gberaudo I agree this is a good discussion to have.  Previously, we said the user should call `map.setTarget(null)` to do a bit of cleanup, but we didn't expose `map.dispose()`.  I don't have time to find the links right now.  But because this pull request doesn't change anything about exports, I think it is best to have that discussion on a fresh ticket.